design/design-doc-template-stacks-experience.md

```diff
@@ -0,0 +1,473 @@
+  now.
+* Setting status in the CRD. This would be nice to have, but it seems
+  independent from the other stuff we're working on. It duplicates
+  information already in the system, so it's not a topmost priority at
+  this time.
+
+Also out of scope is the internal representation of configuration, and
+the implementation of the stack manager. See the [complementary
+internals-oriented Template Stack design doc](https://github.com/crossplaneio/crossplane/blob/master/design/one-pager-template-stacks.md)
+for those details.
+
+## Terms
+
+* A **render request** is an instance of a CRD recognized by the
+  Template Stack. It tells the stack that some configuration is
+  desired.
+* A **template** is a chunk of configuration which will be reused by
+  the Template Stack to create configuration.
+* A **render** is the configuration which is the output of the stack's
+  controller in response to a render request. The configuration is
+  created by processing the stack's input templates using the desired
+  engine, and using the render request for context.
+* **Behaviors** are the configuration settings for a Template Stack.
+  For example, so that its controller knows how to respond to a render
+  request of a known type.
+
+## How to read this document
+
+This document is intended to be read alongside the [template stacks
+experience repository](https://github.com/suskin/template-stack-experience),
+which contains realistic code samples for different pieces of the
+functionality.
+
+Additionally, for more detail about the internals of template stacks on
+the stack manager and Crossplane side, see the [design doc focused on
+this](https://github.com/crossplaneio/crossplane/blob/master/design/one-pager-template-stacks.md)
+
+## Design
+
+The overall flow of authoring and consuming a template stack will be
+very similar for all scenarios. It is expected that template stacks will
+support a wide variety of scenarios, so this section will show some
+examples to help explain them.
+
+For a more complete treatment of example scenarios, see the [template
+stack experience repository](https://github.com/suskin/template-stack-experience).
+
+### Writing
+
+Creating a template stack from scratch involves only a couple steps. The
+project must first be initialized, and then the yamls must be put in a
+particular directory. The rbac requirements for the stack must also be
+configured, though some of that is done by the stack manager at runtime.
+At a high level, the steps would be as follows:
+
+1. Create a project directory.
+2. In the project directory, `kubectl crossplane stack init --template
+   myorg/mystack` to create the boilerplate of the stack layout.
+3. Relative to the project directory, put the yamls in
+   `config/stack/manifests/resources/templates` .
+4. Add CRD definitions; this could be done by putting CRD yaml
+   definitions in `config/stack/manifests/resources` or in
+   `config/crd/bases`.
+5. Add configuration to `stack.yaml` to
```

Comment:
I'm curious about the path to stack.yaml. If the omitted path indicates that this is a topmost file, I think "app.yaml" deserves that position (and the name stack.yaml).

The definition of CRDs -> Templates is a lesser concern than the definition of the Stack itself (which app.yaml provides).

Would stack.yaml be more "at home" alongside the CRDs (with other Stack extending files) or perhaps alongside the template files?

Should stack.yaml be a single file for the entire Stack or could it be split up by CRD as other stack metadata can?